### ISSUE TITLE
Paper results for NarrativeQA

### ISSUE THREAD

**Issue by DbrRoxane** [shmsw25/qa-hard-em#19]
Hi,

I tried to reproduce your results on NarrativeQA over sumaries with the same hyperparameters but I have really lower results (you have 58.1/58.8 and I have 47.07/46.61).
My guess is that the difference could lay in the span extraction method. Would you have your code for span selection? or even better, the dataset?

**Comment by shmsw25** [comment]
Thanks for asking.

I just tried reproducing the number and was actually able to get better numbers than the numbers in the paper (63.7 and 64.0 rouge-l) using `run.sh`. Just to clarify, are your numbers on rouge-l, not F1? In the open-source code, it only prints EM and F1 scores because the open source was mainly for open-domain QA, but the official evaluation metric for NarrativeQA is rouge-l.
In our experience, rouge-l is typically higher than F1 score. (For instance, the result they I have gotten (63.7 and 64.0 in rouge-l) are 47.2 and 47.4 in F1 score.) I used python ROUGE package to compute rouge-l score.

Here is the model checkpoint: [link](https://drive.google.com/file/d/1F_LW73hVMiWaVs2LpnFcQIAqfMatJose/view?usp=sharing)
And this is the data (with distant supervision span labels), which is directly applicable to this open source code: [link](https://drive.google.com/file/d/1xt_aJrMyg0fVVEjUBHyuoW6xIjpqHtNr/view?usp=sharing).

Please let me know if you are still having the problem!

**Comment by DbrRoxane** [comment]
Hi @shmsw25 ,

Yes I use Rouge-L but just inside writing_prediction. should I also change the loss function?  (which is currently CrossEntropy)

Thank you for the data and the checkpoint, it will help a lot :)

**Comment by shmsw25** [comment]
Hi @DbrRoxane, hmm I don't think so. I kept training script that uses F1-score, and just evaluate the final prediction using rouge-l. I wonder if your evaluation script is same as [this](https://github.com/shmsw25/qa-hard-em/issues/18) which uses python rouge package?

**Comment by DbrRoxane** [comment]
Before I used an other package to compute the rouge-l score; I will try now with your script. I keep you in touch, tanks,

In any case, here is the evaluation lines: 
[evaluation script](https://github.com/DbrRoxane/qa-hard-em/blob/49f17a6aa2f4510fa7585c8bab7f311a67f2b19f/evaluation_script.py#L50)
[evaluate qa](https://github.com/DbrRoxane/qa-hard-em/blob/49f17a6aa2f4510fa7585c8bab7f311a67f2b19f/evaluate_qa.py#L232)

**Comment by shmsw25** [comment]
Thanks for the reference. It actually looks like it is using the same package (or maybe you just updated it)? One difference is that we didn't use `normalize_answer`, because `normalize_answer` was part of `exact_match` and `f1_score`.

Would you test with my checkpoint and get the number, just to see if you are able to reproduce 63.7 and 64.0? In that case we could figure out if the issue is in the evaluation side or in the training side.

**Comment by DbrRoxane** [comment]
Funny : with the number of paragraph=5 I have 66.73 on dev set and 66.56 on test set >< with your data and checkpoints
(my predict batch size is 256 but it is not supposed to change something so I guess the difference is from the normalization? )

So I guess I have worst results because we do not have the same way of label the answer /spans. I will check this further :)

**Comment by shmsw25** [comment]
Great to hear! Yes I think batch size shouldn't matter. Please try with the span data in the link above and let me know if it still doesn't work :)

**Comment by DbrRoxane** [comment]
Hi,

Would you actually have the code you used to generate the data? 
We look like tokenizing differently although I use your BasicTokenizer : it looks like you remove punctuation except comas but I cannot find anything that would do that in the code. 
Also I am not sure how and if it is important to split the summaries into different lists (apparently each paragraph is in a different list)

So ja, basically, we have similar annotation but different tokenization and way to split. 

Best,
Roxane

**Comment by DbrRoxane** [comment]
Hi,

@shmsw25 by examinating your data from closer, I have just realized that in your test set, you compare the prediction with the weak labels as ground truth. Should not you compare prediction with the real GT from NarrativeQA ?

**Comment by shmsw25** [comment]
Hi @DbrRoxane,
No, we compared with the real groundtruth from NarrativeQA. We ignored the F1 score and Exact Match score from the training script, as it is not the official metric anyways. If you compute the ROUGE-L score against the real groundtruth from the predictions I attached in this thread, you will be able to get 63.7 and 64.0 in ROUGE-L.
